server/pulp/server/async/app.py

```diff
@@ -72,7 +166,9 @@ def initialize_worker(sender, instance, **kwargs):
     initialization.initialize()
 
     # Delete any potential old state
-    tasks._delete_worker(sender, normal_shutdown=True)
+    existing_worker = Worker.objects(name=sender).first()
+    if existing_worker:
```

Comment:
I see what you're saying, but couldn't a new worker could just as easily start up during the execution of _delete_worker()? 

Speaking of which, would it make sense to move _delete_worker() from tasks.py to worker_utils.py, and get rid of the prepending _, since not an actual celery task, and it's being used in tons of places outside of tasks.py?

The same applies to several other free functions in tasks.py.